core/src/main/java/brooklyn/util/flags/TypeCoercions.java

```diff
@@ -709,6 +710,13 @@ public QuorumCheck apply(final String input) {
                 return QuorumChecks.of(input);
             }
         });
+        registerAdapter(ArrayList.class, String[].class, new Function<ArrayList, String[]>() {
+            @Nullable
+            @Override
+            public String[] apply(@Nullable ArrayList arrayList) {
+                return (String[]) arrayList.toArray(new String[]{});
```

Comment:
If marking as `@Nullable`, then should handle that: return null if `arrayList == null`?